Source/Scene/PointPrimitive.js

```diff
@@ -0,0 +1,488 @@
+     */
+    var PointPrimitive = function(options, pointPrimitiveCollection) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        //>>includeStart('debug', pragmas.debug);
+        if (defined(options.scaleByDistance) && options.scaleByDistance.far <= options.scaleByDistance.near) {
+            throw new DeveloperError('scaleByDistance.far must be greater than scaleByDistance.near.');
+        }
+        if (defined(options.translucencyByDistance) && options.translucencyByDistance.far <= options.translucencyByDistance.near) {
+            throw new DeveloperError('translucencyByDistance.far must be greater than translucencyByDistance.near.');
+        }
+        //>>includeEnd('debug');
+
+        this._show = defaultValue(options.show, true);
+        this._position = Cartesian3.clone(defaultValue(options.position, Cartesian3.ZERO));
+        this._actualPosition = Cartesian3.clone(this._position); // For columbus view and 2D
+        this._color = Color.clone(defaultValue(options.color, Color.WHITE));
+        this._outlineColor = Color.clone(defaultValue(options.outlineColor, Color.TRANSPARENT));
+        this._outlineWidth = defaultValue(options.outlineWidth, 0.0);
+        this._pixelSize = defaultValue(options.pixelSize, 10.0);
+        this._scaleByDistance = options.scaleByDistance;
+        this._translucencyByDistance = options.translucencyByDistance;
+        this._id = options.id;
+        this._collection = defaultValue(options.collection, pointPrimitiveCollection);
+
+        this._pickId = undefined;
+        this._pickPrimitive = defaultValue(options._pickPrimitive, this);
+        this._pointPrimitiveCollection = pointPrimitiveCollection;
+        this._dirty = false;
+        this._index = -1; //Used only by PointPrimitiveCollection
+    };
+
+    var SHOW_INDEX = PointPrimitive.SHOW_INDEX = 0;
+    var POSITION_INDEX = PointPrimitive.POSITION_INDEX = 1;
+    var COLOR_INDEX = PointPrimitive.COLOR_INDEX = 2;
+    var OUTLINE_COLOR_INDEX = PointPrimitive.OUTLINE_COLOR_INDEX = 3;
+    var OUTLINE_WIDTH_INDEX = PointPrimitive.OUTLINE_WIDTH_INDEX = 4;
+    var PIXEL_SIZE_INDEX = PointPrimitive.PIXEL_SIZE_INDEX = 5;
+    var SCALE_BY_DISTANCE_INDEX = PointPrimitive.SCALE_BY_DISTANCE_INDEX = 6;
+    var TRANSLUCENCY_BY_DISTANCE_INDEX = PointPrimitive.TRANSLUCENCY_BY_DISTANCE_INDEX = 7;
+    PointPrimitive.NUMBER_OF_PROPERTIES = 8;
+
+    function makeDirty(pointPrimitive, propertyChanged) {
+        var pointPrimitiveCollection = pointPrimitive._pointPrimitiveCollection;
+        if (defined(pointPrimitiveCollection)) {
+            pointPrimitiveCollection._updatePointPrimitive(pointPrimitive, propertyChanged);
+            pointPrimitive._dirty = true;
+        }
+    }
+
+    defineProperties(PointPrimitive.prototype, {
+        /**
+         * Determines if this pointPrimitive will be shown.  Use this to hide or show a pointPrimitive, instead
```

Comment:
Are you sure about saying `pointPrimitive` throughout the doc?  Why not just `point`?